### PR TITLE
Authentication service/tests

### PR DESCRIPTION
**Pull request by elvincheng3**
## Description/Problem

Closes [Authentication service/tests](https://trello.com/c/sHO1QMgO/56-authentication-controller-service-tests)

Add basic auth groundwork.

## Solution

Add basic auth groundwork.

## Dependencies

[X] This PR adds new dependencies

## Testing

Please describe how you tested this PR (manually and/or automatically)
Provide instructions so we can reproduce, and include screenshots of UI changes.

- Manual Tests?
  - Test out the /auth/login route, and try accessing the protected root route after authenticating with JWT token.
- Unit/Integration/E2E Tests?
  - Added unit tests for auth service

### CODE REVIEW

**Comment by elvincheng3** [comment]
[Authentication service/tests](https://trello.com/c/sHO1QMgO/56-authentication-service-tests)

File: apps/server/src/app.module.ts [hunk 1]
```diff
@@ -24,8 +27,21 @@ import { ConfigModule } from '@nestjs/config';
     ConfigModule.forRoot({
       envFilePath: '.env',
     }),
+    AuthModule,
+    PassportModule,
+    JwtModule.register({
+      secret: process.env.JWT_SECRET,
```
Comment (AnshulShirude) [inline apps/server/src/app.module.ts:33]: What does secret mean?

File: apps/server/src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   */
+  async validateEmployee(
+    email: string,
+    pass: string,
+  ): Promise<EmployeeEntity | null> {
+    const user = await this.employeesService.findOneByEmail(email);
+    if (user?.pswdHash && !(await bcrypt.compare(pass, user.pswdHash!))) {
+      return null;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (AnshulShirude) [inline apps/server/src/auth/auth.service.ts:28]: Is this for eslint? Do we need to keep this line?

**Review by AnshulShirude — APPROVED** [review]
2 small questions are linked, but looks good.

File: apps/server/src/app.module.ts (hunk 1 above)
Comment (elvincheng3) [reply apps/server/src/app.module.ts:33]: it's what's used to generate the jwt token and decode it

File: apps/server/src/auth/auth.service.ts (hunk 2 above)
Comment (elvincheng3) [reply apps/server/src/auth/auth.service.ts:28]: we're basically splitting out the password from the object and using js destructuring to do it, but that means we have an unused variable that eslint complains abt. just telling eslint to ignore this one line for that error

**Pull request merged**